server/src/test/java/org/elasticsearch/common/settings/SettingsTests.java

```diff
@@ -492,6 +492,13 @@ public void testSecureSettingIllegalName() {
         assertTrue(e.getMessage().contains("does not match the allowed setting name pattern"));
     }
 
+    public void testValidateStringSetting() {
+      Settings settings = Settings.builder().putList("foo.bar", Arrays.asList("bla-a", "bla-b")).build();
+      Setting<String> stringSetting = Setting.simpleString("foo.bar");
+      IllegalArgumentException e = expectThrows(IllegalArgumentException.class, () -> stringSetting.get(settings));
+      assertEquals("Found list type value for setting [foo.bar] but did not expect a list for it.", e.getMessage());
+    }
+
     public void testGetAsArrayFailsOnDuplicates() {
```

Comment:
can we also have a happy path testcase where `Setting.listSetting` parses successfully ?